ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/HikariJdbcConnectionProvider.scala

```diff
@@ -29,7 +31,26 @@ object HikariConnection {
       connectionTimeout: FiniteDuration,
       metrics: Option[MetricRegistry],
   ): ResourceOwner[HikariDataSource] =
-    ResourceOwner.forCloseable(() => {
+    new Owner(
+      serverRole,
+      jdbcUrl,
+      minimumIdle,
+      maxPoolSize,
+      connectionTimeout,
+      metrics
+    )
+
+  final class Owner(
```

Comment:
It would probably make sense to move this in its own file and giving it a meaningful name.

Reply:
Or just make `HikariConnection` itself a `ResourceOwner`. I don't think there's a need for the indirection.